rules-reviewed/jboss-eap/websphere/websphere-mq.windup.xml

```diff
@@ -0,0 +1,846 @@
+                <classification title="WebSphere MQ JMS InitialContextFactory">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+                <hint effort="1">
+                    <message>For EAP 6, migrate to JBoss HornetQ.</message>
+                    <link description="Migration Guide - HornetQ" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.4/html-single/Migration_Guide/index.html#Migrate_Your_Application_to_Use_HornetQ_as_the_JMS_Provider"/>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+            <where param="ext"><matches pattern="(java|properties|xml)"/></where>
+        </rule>
+
+        <!-- .scp config files - WINDUPRULE-1 -->
+        <rule id="websphere-mq-04000">
+            <when>
+                <filecontent filename="{*}.scp" pattern="{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ JMS configuration file">
+                </classification>
+                <hint effort="3">
+                    <message>The recommended method of configuring messaging in JBoss EAP 6 is in either the Management Console or Management CLI.</message>
+                    <link description="Configure Messaging with HornetQ" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.4/html-single/Migration_Guide/index.html#Configure_Messaging_with_HornetQ2"/>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+
+
+
+        <rule id="websphere-mq-05000">
+            <when>
+                <javaclass references="com.ibm.msg.client.wmq.WMQConstants"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ Everyplace JMS constants">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+                <hint effort="1">
+                    <message>Replace with Java EE 6 JMS.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="websphere-mq-06000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeJMSAuthenticator"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ Everyplace JMS authentication interface">
```

Comment:
Again, do we really need a classification on every single type reference? This is going to blow up the report. Classifications should be succinct. "WebSphere Proprietary JMS", or something like that at the package level.